#17 user workflow (P2R to Redlink) (#27)
MORE-Platform/more-limesurvey: doc/user-management-and-workflow/user-management-and-workflow.md

--- 

## 2. Operational roles and responsibilities (High level)

For each deployed system (e.g., study environment), a defined set of roles is required to ensure proper user and access management.

The following roles are established:

### System Administrator
- Responsible for configuring and managing user roles within the system
- Can assign and adjust roles for other users
- Acts as the main authority for access control within the system

### User Management Contact
- Responsible for maintaining the shared user-role tracking list
- Coordinates user access requests and updates
- Communicates changes to System Administrator
- This role may be fulfilled by the Study Administrator

### System Users (Researcher / Viewers)
- Standard users of the system
- Have access to studies based on assigned permissions
- Do not manage roles or user configurations

---

## 3. P2R context: Responsibilities

In the P2R context, responsibilities are assigned as follows:

### LBI (User Management and System Users)
- Signs up their users in the self-register process or
- provides accurate user information list via email (see section 4)
- Communicates updates and changes in a timely manner

> Assigned contact: **Gunnar (Study PI)**

Reviewer: ```suggestion
Assigned contact: **Gunnar Treff (Study PI)**
```

 
### Redlink (Study Administrator)
- Acts as System Administrator in the P2R context
- Creates, updates, or deactivates users based on provided input
- Assigns roles according to agreed definitions
- Ensures secure and correct implementation of roles and permissions

> 

Reviewer: ```suggestion

Assigned Contact: **Jan Cortiel** jan.cortiel@redlink.at
```

---

## 4. P2R: User flow

> ### 4.1 Keycloak Roles and Groups

Reviewer: ```suggestion
### 4.1 OAuth Roles and Groups
```

> Keycloak is used for authentication and role assignment. The following groups are defined:

Reviewer: ```suggestion
Keycloak (OAuth) is used for authentication and role assignment. The following groups are defined:
```

| Group               | Realm Roles       | Description                                                                                                                              |
|---------------------|-------------------|------------------------------------------------------------------------------------------------------------------------------------------|
| MORE Administrator  | incl. MORE Admin  | (System Administrator, Platform Administrator): Rights to manage users, emergency functions, no rights to see data or manipulate studies. |
| MORE Researcher     | incl. MORE Viewer | Can create new studies and access existing studies (based on assigned study-level roles)                                                 |
**Default role:** All newly created users are assigned MORE Researcher unless explicitly specified otherwise.

> **Note:** The User Roles and Groups provided by Keycloak are not 1:1 corresponding with the roles inside the MORE platform (Study Administrator, Study Operator, Study Viewer). 

Reviewer: ```suggestion
**Note:** The above Roles are system wide and different to the study roles (Study Administrator, Study Operator, Study Viewer). The system wide role MORE Researcher can be assigned to any study role. This assignment is done by the study configuration.
```

---

### 4.2 User Provisioning Options

#### 1. Self-registration
- LBI registers thier Users directly via the Studymanager self sign-up process
- Default role: MORE Researcher
- No further action required unless role changes are needed

#### 2. Providing a list of users
- LBI may provide a user list to Redlink for provisioning and updates.
- Redlink will create or update users accordingly.

> 

Reviewer: I would like a decision not just the options. Make a suggestion and lets discuss this with LBI

Author: @westei - so yes. the easist way would be a self-register and we are good to go -> then they would only need us if they want to deactivate or delete a user -> how about

1. Self-registration (stays the same)

2. User Lifecycle Management (Create / Update / Deactivate)
- User creation is handled exclusively via self-registration in the system.
- If LBI needs to update a user’s role (other than the default Study Researcher), deactivate, or delete a user, they must provide Redlink with a user list specifying the required changes (e.g. update role to X, deactivate, delete).
- Redlink will process the requested changes and apply them accordingly in the system.

that would be my preferable workflow. so they can independently handle their user registration with the default role. and we'll support with specific changes.

Reviewer 2: @westei @iaigner We could also create a new Keycloak User admin account for LBI, so they can manage their users themselfs. This excludes control over any other setting in keycloak

Author: if that's possible that would give them more independence. i like that idea.

Reviewer: @janoliver20 I will add this as an option to the document

---

### 4.3 Required information for role assignment 

The following information must be provided in the user list. The list is sent via email to Redlink for processing.

| Field                     | Description                                | Required |
| ------------------------- |--------------------------------------------|----------|
| E-mail                    | Email address of the user                  | Yes      |
| System Role(s) / Group(s) | Role(s) assigned to the user               | No       |
| Action type               | Type of change (new / update / deactivate) | Yes      |

> 

Reviewer: I think we need also the SSH key of the user - @janoliver20 please comment

Author: I asked Jan yesterday and he said if they already registered or had a self-sign in the email is sufficient. -> @janoliver20 ?

Reviewer 2: @westei No ssh key necessary, only the email address

| Group               | Realm Roles       | Description                                                                                                                              |
|---------------------|-------------------|------------------------------------------------------------------------------------------------------------------------------------------|
| MORE Administrator  | incl. MORE Admin  | (System Administrator, Platform Administrator): Rights to manage users, emergency functions, no rights to see data or manipulate studies. |
| MORE Researcher     | incl. MORE Viewer | Can create new studies and access existing studies (based on assigned study-level roles)                                                 |
**Default role:** All newly created users are assigned MORE Researcher unless explicitly specified otherwise.

**Note:** The above Roles are system wide and different to the study roles (Study Administrator, Study Operator, Study Viewer). The system wide role MORE Researcher can be assigned to any study role. This assignment is done by the study configuration.

---

### 4.2 User Provisioning Options

#### 1. Self-registration

> - LBI registers thier Users directly via the Studymanager self sign-up process

Reviewer 2: ```suggestion
- LBI registers their Users directly via the Studymanager self sign-up process
```
